frontend/packages/ceph-storage-plugin/src/components/disk-inventory/ocs-disks-list.tsx

```diff
@@ -189,24 +189,26 @@ const OCSDisksList: React.FC<TableProps> = React.memo((props) => {
     }
 
     if (tiLoaded && !tiLoadError && tiData.length) {
-      const newData: ReplacementMap = tiData.reduce((data: ReplacementMap, ti) => {
+      const newData: ReplacedDisk[] = tiData.reduce((data: ReplacedDisk[], ti) => {
         const { devicePath, deviceID, deviceOsd, deviceNode, deviceSerial } =
           getAnnotations(ti) || {};
         if (devicePath && deviceOsd && deviceNode === nodeName) {
-          data[devicePath] = {
+          data.push({
             osd: deviceOsd,
             disk: {
               id: deviceID,
+              path: devicePath,
               serial: deviceSerial,
+              node: nodeName,
             },
             status: getTiBasedStatus(ti.status.conditions?.[0].type),
-          };
+          });
         }
         return data;
-      }, {});
-      if (!_.isEqual(newData, ocsState.replacementMap)) {
+      }, []);
+      if (!_.isEqual(newData, ocsState.replacedDiskList)) {
```

Comment:
Even one disk is not matching for a node then we are considering everything as replaced disk?

Reply:
is one have only one disk at a time, that why we are keeping condition like this?

Reply:
This list builds the details of replaced disk on the given node.
We collate all replaced disks data and then pass it down to `OCSStatus` component where per disk is rendered.
If a disk is not for a specific node, then yes we dont want that in this list.

I hope I understand the concern and answered correctly.

Reply:
ack